bundle/config/mutator/log_resource_references.go

```diff
@@ -0,0 +1,145 @@
+package mutator
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/dynvar"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/cli/libs/structaccess"
+)
+
+// Longest field name:
+// >>> len('hard_deletion_sync_min_interval_in_seconds') == 42
+const maxFieldLength = 42
+
+const maxMetricLength = 200
+
+// logResourceReferences scans resources for ${resources.*} references and logs them.
+func LogResourceReferences() bundle.Mutator {
+	return &logResourceReferences{}
+}
+
+type logResourceReferences struct{}
+
+func (m *logResourceReferences) Name() string {
+	return "LogResourceReferences"
+}
+
+func (m *logResourceReferences) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	err := b.Config.Mutate(func(root dyn.Value) (dyn.Value, error) {
+		pattern := dyn.NewPattern(dyn.Key("resources"))
+		used := map[string]struct{}{}
+		_, err := dyn.MapByPattern(root, pattern, func(p dyn.Path, v dyn.Value) (dyn.Value, error) {
```

Comment:
You don't need to call `MapByPattern` here. You can instead use `b.b.Config.Value().Get("resources")` and call WalkReadOnly on that.